odoo/osv/expression.py

```diff
@@ -727,6 +727,9 @@ def to_ids(value, comodel):
             elif value and isinstance(value, (tuple, list)) and all(isinstance(item, pycompat.string_types) for item in value):
                 names = value
             elif isinstance(value, pycompat.integer_types):
+                if value is False:
+                    _logger.warning("Unexpected domain [%s], interpreted as False", leaf)
+                    return []
```

Comment:
Maybe add a comment with a bit of rationale for this choice, e.g. something like:
``` 
# Cheaper to treat the domain as False than to make it match everything,
# given this nonsensical domain
```